evm/consensus/pow.py

```diff
@@ -65,3 +65,18 @@ def check_pow(block_number: int,
             encode_hex(mining_output[b'mix digest']), encode_hex(mix_hash)))
     result = big_endian_to_int(mining_output[b'result'])
     validate_lte(result, 2**256 // difficulty, title="POW Difficulty")
+
+
+MAX_TEST_MINE_ATTEMPTS = 1000
+
+
+def test_mine(block_number, mining_hash, difficulty):
```

Comment:
Coincidentally I implemented the same method when I started writing the "Understanding the mining process" guide. I think it's beneficial for have this method even if only being useful for tests and docs.

 I would however propose that we rename it to something like `mine_nonce` or maybe even `mine_nonce_and_mixhash`. Thoughts?

Reply:
I'd suggest adding `pow` or `ethash` to the name to make it clear that it's proof-of-work mining.

Reply:
I like `mine_pow_nonce`.

Reply:
sgtm :+1:

Reply:
Currently the callsites import the `pow` package, so this is always called like `pow.test_mine()`, which is why I didn't want to include `pow` in the function's name. I can rename if you guys feel strong about it

Reply:
It's still prefixed with `test` though.

Reply:
Correct, because it is not meant to be used in production code as it performs only a small number of attempts and hence will never actually work on anything other than dummy blocks with a ridiculously low difficulty.

Reply:
Sure. Might be just me but the `test` prefix suggests to me that this method performs a test which it doesn't.

Reply:
It will also cause `pytest` to pick up and try to run the function (if it's not run against the `tests` directory).

Reply:
I renamed it as part of the other PR that is about to land https://github.com/ethereum/py-evm/pull/641/commits/9a3cfdac75b842378f108f75283c5ecaf2d69dae